File: 0026-ipv4-validator/ipv4-validator.js
function isValidIPv4(ipv4) {
    if (ipv4.split('.').length - 1 !== 3) {
        return false;
    }
    
    const parts = ipv4.split('.');
    
    if (parts.length !== 4) {
        return false;
    }
    
    for (let part of parts) {
        if (!part) {
            return false;
        }
        
        if (part.length > 1 && part[0] === '0') {
            return false;
        }
        
        if (!/^\d+$/.test(part)) {
            return false;
        }
        
        const num = parseInt(part, 10);
        
        if (num < 0 || num > 255) {
            return false;
        }
    }
    
    return true;
}

File: 0026-ipv4-validator/ipv4_validator.py
"""
Given a string, determine if it is a valid IPv4 Address. A valid IPv4 address consists of 
four integer numbers separated by dots (.). Each number must satisfy the following conditions:

It is between 0 and 255 inclusive.
It does not have leading zeros (e.g. 0 is allowed, 01 is not).
Only numeric characters are allowed.

"""

def is_valid_ipv4(ipv4):
    if ipv4.count('.') != 3:
        return False
    
    parts = ipv4.split('.')
    
    if len(parts) != 4:
        return False
    
    for part in parts:
        if not part:
            return False
        
        if len(part) > 1 and part[0] == '0':
            return False
        
        if not part.isdigit():
            return False
        
        num = int(part)
        
        if num < 0 or num > 255:
            return False

    return True

    